# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ANEMBHARGAV/flyrank_ml_intern_w1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### My lane: Freestyle — Content Refresh Prioritization

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Lane selected: Freestyle — Content Refresh Prioritization")

Lane selected: Freestyle — Content Refresh Prioritization


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision, action, and cost

The decision is which pages should be reviewed or refreshed first. A content or SEO team could use the analysis to prioritize a limited number of pages for manual review. A wrong recommendation could cause the team to spend time refreshing a page that does not need attention, while missing a page that may actually need review. Therefore, the result should be used as decision-support rather than an automatic decision.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Decision: prioritize pages for manual content review")

Decision: prioritize pages for manual content review


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import os
import pandas as pd
import subprocess
import sys

REPO_URL = "https://github.com/ANEMBHARGAV/flyrank_ml_intern_w1"
REPO_DIR = "flyrank_ml_intern_w1"

if not os.path.exists(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Declining rate:", round(df["trend_direction"].str.lower().eq("down").mean(), 3))
print("Median impressions over 90 days:", round(df["impressions_90d"].median(), 1))

Number of pages: 30000
Declining rate: 0.542
Median impressions over 90 days: 731.0


### What the numbers show

The dataset contains 30,000 pages, and about 54.2% of pages are labeled as declining based on the starter definition. The median impressions over 90 days is 731, showing that pages have different levels of search exposure. These observations make it useful to investigate which observable signals can help prioritize pages for review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can and cannot claim

This work can identify observed patterns in the available search and content data and provide directional decision-support for prioritizing pages for review. It can measure relationships between page characteristics and the observed declining label. It cannot prove that one factor caused a page's performance to change, and it cannot predict or explain Google's ranking algorithm. The results should therefore be treated as measured evidence for prioritization, not causal proof.

In [7]:
print("Claim type: observed, directional, and decision-support")

Claim type: observed, directional, and decision-support


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.